code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -1 +1,177 @@
+
+	succeed_activate()
+	if(!do_after(X, delay, TRUE, null, BUSY_ICON_BUILD))
+		if(!QDELETED(src))
+			to_chat(X, "<span class='xenodanger'>We abort transferring our consciousness, expending our precious plasma for naught.</span>")
+			return fail_activate()
+
+	if(!check_build_location(target_turf, X)) //Check target turf for suitability after completion
+		return fail_activate()
+
+	core.forceMove(get_turf(target_turf)) //Move the core
+	core.obj_integrity = core.max_integrity //Reset the core's health
+	to_chat(owner, "<span class='xenodanger'>We succeed in transferring our consciousness to a new neural core!</span>")
+
+	add_cooldown()
+
+	GLOB.round_statistics.hivemind_reposition_core_uses++ //Increment the statistics
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hivemind_reposition_core_uses")
+
+	if(!is_centcom_level(core))
+		var/area/core_new_area = get_area(core)
+		xeno_message("Hive: \The [X] has <b>transferred its core</b>[A? " to [sanitize(core_new_area.name)]":""]!", 3, X.hivenumber) //Let the hive know the Hivemind's new location.
+
+/datum/action/xeno_action/activable/reposition_core/proc/check_build_location(turf/target_turf, mob/living/carbon/xenomorph/hivemind/X)
+
+	if(!X || !target_turf) //Sanity
+		return FALSE
+
+	if(target_turf.density) //Check to see if there's room
+		to_chat(X, "<span class='xenodanger'>We require adequate room to transfer our core.</span>")
+		return FALSE
+
+	var/obj/effect/alien/weeds/W = locate() in range(0, target_turf) //Make sure we actually have weeds at our destination.
+	if(!W)
+		to_chat(X, "<span class='xenodanger'>There are no weeds for us to transfer our consciousness to!</span>")
+		return FALSE
+
+	for(var/obj/structure/O in target_turf.contents)
+		if(O.density && !(O.flags_atom = ON_BORDER))
+			to_chat(X, "<span class='xenodanger'>An object is occupying this space!</span>")
+			return FALSE
+
+	return TRUE
+
+
+
+// ***************************************
+// *********** Mind Wrack
+// ***************************************
+/datum/action/xeno_action/activable/mind_wrack
+	name = "Mind Wrack"
+	action_icon_state = "screech"
+	mechanics_text = "Afflicts the target's mind with disorienting and painful psychic energy. Effect is greater the closer the target is to your core, and the more xenos that are within 3 tiles when used."
+	ability_name = "mind wrack"
+	plasma_cost = 100
+	keybind_signal = COMSIG_XENOABILITY_MIND_WRACK
+	cooldown_timer = 30 SECONDS
+
+/datum/action/xeno_action/activable/mind_wrack/on_cooldown_finish()
+	owner.playsound_local(owner.loc, 'sound/voice/alien_drool1.ogg', 50, TRUE)
+	to_chat(owner, "<span class='xenonotice'>We regain the strength to assault the minds of our enemies.</span>")
+	return ..()
+
+
+/datum/action/xeno_action/activable/mind_wrack/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+	var/mob/living/carbon/victim = A
+
+	if(!istype(victim) )
```

Comment:
same here, use can_use_ability() and the ability flags for dead mobs etc

Reply:
This is done.